src/vs/gitpod/browser/workbench/workbench.html

```diff
@@ -3,6 +3,74 @@
 <html>
 	<head>
 		<script>
+			async function gitpodMetricsAddCounter(metricsName, labels, value) {
+				function getMetricsUrl() {
+					const baseWorkspaceIDRegex = "(([a-f][0-9a-f]{7}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})|([0-9a-z]{2,16}-[0-9a-z]{2,16}-[0-9a-z]{8,11}))";
+					// this pattern matches URL prefixes of workspaces
+					const workspaceUrlPrefixRegex = RegExp(`^([0-9]{4,6}-)?${baseWorkspaceIDRegex}\\.`);
+					let url = new URL(window.location.href)
```

Comment:
suggestion (nit): we don't reinitialize / reassign `url`, so we can make it a constant

```suggestion
					const url = new URL(window.location.href)
```